arangod/RestHandler/RestAdminClusterHandler.cpp

```diff
@@ -804,6 +808,148 @@ RestStatus RestAdminClusterHandler::handleQueryJobStatus() {
+          return RestStatus::DONE;
+        } else if (path[2] != "Pending" && path[2] != "ToDo") {
+          generateError(Result{400, path[2] + " jobs can no longer be cancelled."});
+          return RestStatus::DONE;
+        }
+
+        auto sendTransaction = [&] {
+          VPackBuffer<uint8_t> trxBody;
+          { VPackBuilder builder(trxBody);
+            { VPackArrayBuilder trxs(&builder);
+              if (path[2] == "ToDo") {
+                VPackArrayBuilder trx(&builder);
+                { VPackObjectBuilder op(&builder);
+                  builder.add("arango/Target/ToDo/" + jobId + "/abort", VPackValue(true)); }
+                { VPackObjectBuilder pre(&builder);
+                  builder.add(VPackValue("arango/Target/ToDo/" + jobId));
+                  { VPackObjectBuilder val(&builder);
+                    builder.add("oldEmpty", VPackValue(false)); }}
+              }
+              VPackArrayBuilder trx(&builder);
+              { VPackObjectBuilder op(&builder);
+                builder.add("arango/Target/Pending/" + jobId + "/abort", VPackValue(true)); }
+              { VPackObjectBuilder pre(&builder);
+                builder.add(VPackValue("arango/Target/Pending/" + jobId));
+                { VPackObjectBuilder val(&builder);
+                  builder.add("oldEmpty", VPackValue(false)); }}
+            }
+          }
+          return AsyncAgencyComm().sendWriteTransaction(60s, std::move(trxBody));
+        };
+
+        waitForFuture(
+          sendTransaction()
+          .thenValue([this](AsyncAgencyCommResult&& wr) {
+            if (!wr.ok()) {
+              if (wr.statusCode() == 412) {
+                auto results = wr.slice().get("results");
+                if (results[0].getNumber<int64_t>() == 0 &&
+                    results[1].getNumber<int64_t>() == 0) {
+                  generateError(Result{TRI_ERROR_HTTP_PRECONDITION_FAILED,
+                                       "Job is no longer pending or to do"});
+                }
+              } else {
+                generateError(wr.asResult());
+              }
+            }
+            return futures::makeFuture();
+          })
+          .thenError<VPackException>([this](VPackException const& e) {
+            generateError(Result{TRI_ERROR_HTTP_SERVER_ERROR, e.what()});
+          })
+          .thenError<std::exception>([this](std::exception const& e) {
+            generateError(rest::ResponseCode::SERVER_ERROR,
+                          TRI_ERROR_HTTP_SERVER_ERROR, e.what());
+          }));
+        VPackBuffer<uint8_t> payload;
+        {
+          VPackBuilder builder(payload);
+          VPackObjectBuilder ob(&builder);
+          builder.add("job", VPackValue(jobId));
+          builder.add("status", VPackValue("aborted"));
+          builder.add("error", VPackValue(false));
+        }
+        resetResponse(rest::ResponseCode::OK);
+        response()->setPayload(std::move(payload));
+        return RestStatus::DONE;
```

Comment:
As soon as the transaction was sent the rest handler returns with `OK`. I bet it returns `OK` even if all agents are dead.

Reply:
what do you expect. just adopted to behaviour of the remaining code here.